src/get-variables-functions.ts

```diff
@@ -64,27 +64,27 @@ function getArrayFunction(arrayNode: VariableDeclarationNode): ArrayStateVariabl
   // Name of the array
   const arrayName: string = arrayNode.name;
 
-  // compile param type
-  const paramType: string = typeFix(arrayNode.typeDescriptions.typeString).replace(/contract |struct |enum /g, '');
+  // Array type
+  const arrayType: string = typeFix(arrayNode.typeDescriptions.typeString).replace(/contract |struct |enum /g, '');
 
-  // struct flag
-  const isStruct: boolean = arrayNode.typeName.baseType.typeDescriptions.typeString.includes('struct ');
-
-  // compile base type
+  // Base type
   const baseType: string = typeFix(arrayNode.typeName.baseType.typeDescriptions.typeString).replace(/contract |struct |enum /g, '');
 
+  // Struct flag
+  const isStruct: boolean = arrayNode.typeDescriptions.typeString.includes('struct ');
+
   // If the array is internal we don't create mockCall for it
   const isInternal: boolean = arrayNode.visibility == 'internal';
 
   const arrayStateVariableFunctions: ArrayStateVariableOptions = {
     setFunction: {
       functionName: arrayName,
-      paramType: paramType,
+      arrayType: arrayType,
       paramName: arrayName,
```

Comment:
`paramName` property could be easily discarded, as in practice it only carries redundant information.